src/styles/blocks/calendar.scss

```
@@ -0,0 +1,55 @@
+.calendar {
+  @extend %flex-container;
+  max-width: getWidth($cell-size, $numbers-of-cells, $indent, $cell-gap);
+  padding: 10px;
+  gap: 1px;
+
+  @for $index from 28 through 31 {
+    &--month-length--#{$index} > &__day:nth-child(n + #{$index + 1}) {
+      display: none;
+    }
+  }
```

Не повышай специфичность, задай это сразу на элемент
```suggestion
  @for $index from 28 through 31 {
    .calendar__day:nth-child(n + #{$index + 1}) {
      display: none;
    }
  }
``` 
Исправишь для своего элемента с вложенностью (&)